packages/gatsby-plugin-utils/src/test-plugin-options-schema.ts

```diff
@@ -0,0 +1,35 @@
+import Joi from "joi"
+import { ObjectSchema } from "./utils/plugin-options-schema-joi-type"
+
+type PluginSchemaFunction = ({ Joi }) => ObjectSchema
```

Comment:
I think you can import this type from the `gatsby` package? I added it to the `GatsbyNode` type there!

```JS
import { GatsbyNode } from "gatsby"

GatsbyNode["pluginOptionsSchema"]
```